csharp/src/Google.Protobuf/FieldMaskTree.cs

```diff
@@ -293,6 +293,13 @@ public void Merge(IMessage source, IMessage destination, FieldMask.MergeOptions
                         field.Accessor.SetValue(destination, destinationField);
                     }
 
+                    if (sourceField == null)
+                    {
+                        // If the message field is not present in the source but is in the destination, skip recursing
+                        // so we just keep the destination value recursively.
```

Comment:
I'm not sure what this line means. Perhaps:

```csharp
// If the message field is not present in the source but is in the destination, skip recursing.
// We leave the destination as it is, and move on to the next field to merge.
```

Reply:
This got updated with new behavior and a new comment